MatchPy_Dependency.md

```diff
@@ -0,0 +1,137 @@
+# Make SymPy depend on MatchPy
+
+**Author** Francesco Bonazzi, <franz.bonazzi@gmail.com>
+
+**Status** Draft
+
+**Created** 2021-10-04
+
+**Type** Standards Track
+
+
+## Introduction
+
+MatchPy is a powerful state-of-the-art library for associative-commutative
+pattern matching.  It is written entirely in Python and is licensed under MIT.
+Its API is remotely inspired to Wolfram Mathematica's pattern matcher, although
+there are some differences.
+
+The most useful capability of MatchPy is the ability to perform multiple
+pattern matches at the same time. Once the rules are loaded into memory, a data
+structure is created that efficiently regroups all the pattern matching rules.
+
+Furthermore, MatchPy has also a code generator that can create Python code with
+the decision tree corresponding to the selected set of pattern matching rules.
+
+Benchmarks run on RUBI (Rule-based integration) have shown that even large sets
+of about 10 000 patterns can be run in extremely fast times, once all rules are
+loaded. In particular, MatchPy was able to beat Wolfram Mathematica in speed on
+large sets of pattern matching rules.
```

Comment:
Can you provide some actual timing comparisons (or point to them) so that we can see what there is to gain in a specific way?